smart_contracts/contracts/explorer/faucet/src/lib.rs

```diff
@@ -1,39 +1,538 @@
+            FaucetError::MissingTimeInterval,
+            FaucetError::InvalidTimeInterval,
+        );
+        storage::write(time_interval_uref, new_time_interval);
+    }
+
+    if let Some(new_available_amount) = get_optional_named_arg_with_user_errors::<U512>(
+        ARG_AVAILABLE_AMOUNT,
+        FaucetError::MissingAvailableAmount,
+        FaucetError::InvalidAvailableAmount,
+    ) {
+        let available_amount_uref = get_uref_with_user_errors(
+            AVAILABLE_AMOUNT,
+            FaucetError::MissingAvailableAmount,
+            FaucetError::InvalidAvailableAmount,
+        );
+        storage::write(available_amount_uref, new_available_amount);
+    }
+
+    if let Some(new_distributions_per_interval) = get_optional_named_arg_with_user_errors::<u64>(
+        ARG_DISTRIBUTIONS_PER_INTERVAL,
+        FaucetError::MissingDistributionsPerInterval,
+        FaucetError::InvalidDistributionsPerInterval,
+    ) {
+        let distributions_per_interval_uref = get_uref_with_user_errors(
+            DISTRIBUTIONS_PER_INTERVAL,
+            FaucetError::MissingDistributionsPerInterval,
+            FaucetError::InvalidDistributionsPerInterval,
+        );
+        let remaining_requests_uref = get_uref_with_user_errors(
+            REMAINING_REQUESTS,
+            FaucetError::MissingRemainingRequests,
+            FaucetError::InvalidRemainingRequests,
+        );
+
+        storage::write(
+            distributions_per_interval_uref,
+            new_distributions_per_interval,
+        );
+        // remaining requests == distributions per interval.
+        storage::write(
+            remaining_requests_uref,
+            U512::from(new_distributions_per_interval),
+        );
+    }
+}
+
+#[no_mangle]
+pub fn authorize_to() {
+    let installer = get_account_hash_with_user_errors(
+        INSTALLER,
+        FaucetError::MissingInstaller,
+        FaucetError::InvalidInstaller,
+    );
+
+    if runtime::get_caller() != installer {
+        runtime::revert(FaucetError::InvalidAccount);
+    }
+
+    let authorized_account_public_key = get_optional_named_arg_with_user_errors::<PublicKey>(
+        ARG_TARGET,
+        FaucetError::MissingAuthorizedAccount,
+        FaucetError::InvalidAuthorizedAccount,
+    );
 
-const ARG_TARGET: &str = "target";
-const ARG_AMOUNT: &str = "amount";
+    let authorized_account_uref = get_uref_with_user_errors(
+        AUTHORIZED_ACCOUNT,
+        FaucetError::MissingAuthorizedAccount,
+        FaucetError::InvalidAuthorizedAccount,
+    );
 
-#[repr(u32)]
-enum CustomError {
-    AlreadyFunded = 1,
+    storage::write(authorized_account_uref, authorized_account_public_key);
 }
 
-/// Executes token transfer to supplied account hash.
-/// Revert status codes:
-/// 1 - requested transfer to already funded account hash.
 #[no_mangle]
 pub fn delegate() {
-    let account_hash: AccountHash = runtime::get_named_arg(ARG_TARGET);
+    let id = get_optional_named_arg_with_user_errors(
```

Comment:
There's finite amount of ARG_ or storage constant. I think having functions for each arg (i.e. `get_id_arg() -> Option<u64>` for args or `read_xxx()`/`write_xxx()` for storage etc) could help decrease amount of boilerplate here